app/src/main/res/layout/fragment_wait_times.xml

```diff
@@ -41,16 +40,240 @@
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:fontFamily="sans-serif-medium"
-            android:layout_marginBottom="9dp"
             android:text="Popular Times"
             android:textColor="@color/primary"
             android:textSize="18sp"/>
 
+        <TextView
+            android:id="@+id/wait_time_show_button"
+            android:clickable="true"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_alignTop="@id/wait_time_label"
+            android:layout_alignParentRight="true"
+            android:fontFamily="sans-serif-medium"
+            android:text="Hide"
+            android:textColor="@color/secondary"
+            android:textSize="18sp"/>
+
         <com.github.mikephil.charting.charts.BarChart
             android:id="@+id/wait_time_chart"
             android:layout_below="@id/wait_time_label"
+            android:animateLayoutChanges="true"
+            android:layout_width="match_parent"
+            android:layout_height="156dp"
+            android:layout_marginTop="7dp"
+            android:layout_marginLeft="15dp"
+            android:layout_marginRight="15dp"/>
+
+        <View
+            android:id="@+id/wait_time_x_axis_line"
+            android:animateLayoutChanges="true"
             android:layout_width="match_parent"
-            android:layout_height="156dp"/>
+            android:layout_height="2dp"
+            android:background="@color/inactive"
+            android:layout_below="@id/wait_time_chart"
+            android:layout_marginLeft="15dp"
+            android:layout_marginRight="15dp"/>
+
+        <TableRow
```

Comment:
please create this dynamically in a java class. it'll be much cleaner

Reply:
now that i've looked at this file closer, it's fine to keep this all in the xml file. however, i am certain that there's a better way to create this layout with only one or two `RelativeLayout`s. 

in general, using these layouts is expensive so you want to avoid creating too many per xml file. instead of a `TableRow` you can consider using a singular `LinearLayout` with padding + margins, or in general, replace the `View`s with margins/paddings

Reply:
Are linear layouts better here than relative layouts?

Reply:
@connorreinhold did we resolve this offline?

Reply:
Updated to use two linear layouts, and removed all sub-relative layouts.